src/operators/switchMapFirst.ts

```diff
@@ -0,0 +1,86 @@
+import Operator from '../Operator';
+import Observable from '../Observable';
+import Subscriber from '../Subscriber';
+import tryCatch from '../util/tryCatch';
+import { errorObject } from '../util/errorObject';
+import OuterSubscriber from '../OuterSubscriber';
+import subscribeToResult from '../util/subscribeToResult';
+
+export default function switchMapFirst<T, R, R2>(project: (value: T, index: number) => Observable<R>,
+                                                 resultSelector?: (outerValue: T,
+                                                                   innerValue: R,
+                                                                   outerIndex: number,
+                                                                   innerIndex: number) => R2): Observable<R> {
+  return this.lift(new SwitchMapFirstOperator(project, resultSelector));
+}
```

Comment:
Seems this declaration can be updated to remove `R2` comparing to operator declaration : 
`switchMapFirst?: <R>(project: ((x: T, ix: number) => Observable<any>), projectResult?: (x: T, y: any, ix: number, iy: number) => R) => Observable<R>;`